### DISCUSSION TITLE
Druid being discontinued

### DISCUSSION THREAD

**Discussion by cameronraw** [jpochyla/psst#423] [Ideas]
I was wondering if any thought had been given to an alternative to handling the UI has been considered as Druid is discontinued and a couple of issues I've seen seem to hinge on how Druid handles click events, vectors etc.. 

According to the Druid repo, they're moving over to Xilem - outside of that, the Tauri framework looks like it's come a long way and is even edging closer towards native mobile applications from the same codebase.

I imagine it would be a big undertaking but I was just curious to know if there's been any discussion around this. Thanks!

**Comment by jpochyla** [comment]
Hi, I'd love to port over to Xilem once it's a bit more stable and I have a bit more time, ~end of this year I hope. That said, the core part of Psst needs a lot of work as well.

**Reply by cameronraw** [reply]
Cool, thanks.

**Comment by Insprill** [comment]
Xilem is looking good. My main concern with it would be support. It'd be unfortunate to build an entirely new UI just for it to meet Druid's fate and end up with outstanding bugs that never get fixed, causing things to be broken for us.

Iced has been moving fast and has the backing of System76, which is nice, but their documentation is basically non-existent. From what I've seen, it works pretty differently from Druid too, so it'd be a learning curve.

Tauri looks cool, and I see the use case, but it's a bit overkill for our needs.

If we're rewriting the frontend from scratch, that'd also be an excellent time to look into using librespot for a backend. It was partially discussed in #204 and #359. Joining forces would allow us to improve downstream projects with librespot changes, fix multiple outstanding psst bugs, and add support for things like Spotify Connect.
